traveltimepy/dto/transportation.py

```
@@ -26,6 +29,15 @@ class Cycling(BaseModel):
 class Ferry(BaseModel):
     type: Literal["ferry", "cycling+ferry", "driving+ferry"] = "ferry"
     boarding_time: Optional[int] = None
+    traffic_model: Optional[DrivingTrafficModel] = None
+
+    @model_validator(mode="after")
```

I don't like this part, but see no other way to avoid breaking changes.

---

to be quite frank, it's not 100% necessary to have this validation here, you can let our API handle the error. But it seems fine to me, so it's up to you if you want to keep the validation or not

---

But why do you even have the `traffic_model` param on `Driving` mode for example, if it can only be applied to `driving+ferry`? Or am I misunderstanding smth

---

Ahhh okay nevermind, I see what's going on here

---

> let our API handle the error

Api will be silent about it, optional fields are ignored. I prefer to leave validator, throwing this Exception makes more sense than silently ignore traffic_model param.

---

hmmm interesting, didn't know this. Fair enough then, we can keep the validator